pkg/target/icd.go

```diff
@@ -21,10 +21,8 @@ func (t *Target) WillApplyEle(tag combat.ICDTag, grp combat.ICDGroup, char int)
 
 	val := t.icdTagCounter[char][tag]
 	//increment the counter
-	//if counter > length, then reset back to 0
-	t.icdTagCounter[char][tag]++
-	if t.icdTagCounter[char][tag] == len(combat.ICDGroupEleApplicationSequence[grp]) {
-		t.icdTagCounter[char][tag] = 0
+	if t.icdTagCounter[char][tag] < len(combat.ICDGroupEleApplicationSequence[grp]) - 1 {
+		t.icdTagCounter[char][tag]++
 	}
```

Comment:
If I understand this issue correctly, we want `combat.ICDGroupEleApplicationSequence[grp][val]` to be `0` when `t.icdTagCounter[char][tag] >= len(combat.ICDGroupEleApplicationSequence[grp])`. Current logic will have last value in the sequence be used, which in some cases is not `0`.